parser/lexer.go

```diff
@@ -443,6 +443,12 @@ func startWithNumber(s *Scanner) (tok int, pos Pos, lit string) {
 	if ch0 == '.' || ch0 == 'e' || ch0 == 'E' {
 		return s.scanFloat(&pos)
 	}
+
+	// Identifiers may begin with a digit but unless quoted may not consist solely of digits.
```

Comment:
check "ch0 != unicode.ReplacementChar" because peek() will return unicode.ReplacementChar if it meet EOF, so that "123\EOF" won't be recognized as identifier

To tell the truth, it's not safe to use unicode.ReplacementChar as peek() return EOF, here it is just because the old lexer do it this way...

--------

I've add a test case:
```
create database `123`;
```

it's a valid statement...